Fix -Wpreferred-type-bitfield-enum-conversion warnings in WebCore

ddkilzer opened the pull request:
#### 5a74e184fd2204750d9870b60555fb2f7fb987c8
<pre>
Fix -Wpreferred-type-bitfield-enum-conversion warnings in WebCore
&lt;<a href="https://bugs.webkit.org/show_bug.cgi?id=297439">https://bugs.webkit.org/show_bug.cgi?id=297439</a>&gt;
&lt;<a href="https://rdar.apple.com/158245532">rdar://158245532</a>&gt;

Reviewed by Brent Fulgham.

Warnings found by -Wpreferred-type-bitfield-enum-conversion indicate
that a bitfield used to store an enum type are too small to hold one or
more values defined in the enum.

* Source/WebCore/css/SelectorChecker.cpp:
(WebCore::SelectorChecker::match const):
- Switch to use new definition of PublicPseudoIdMask.
* Source/WebCore/rendering/style/FillLayer.h:
(WebCore::FillLayer::FillBoxBitWidth): Add.
(WebCore::FillLayer::m_clip):
(WebCore::FillLayer::m_origin):
(WebCore::FillLayer::m_clipMax):
- Define FillBoxBitWidth constant for bitwidth of FillBox enum, then
  use it when definining bitfileds of type FillBox.  This fixes the
  warning and a real bug where m_origin and m_clipMax could not hold all
  values of FillBox.
* Source/WebCore/rendering/style/RenderStyleConstants.h:
(WebCore::PseudoId::PublicPseudoIdMask): Remove.
(WebCore::PublicPseudoIdMask): Add.
- Define PublicPseudoIdMask outside enum class PseudoId since it&apos;s too
  large to be held by the RenderStyle::pseudoElementType bitfield.  This
  was not a bug in practice, but fixes the warning by moving the mask
  value outside of the enum itself.
* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::RenderStyle::NonInheritedFlags::hasAnyPublicPseudoStyles const):
- Switch to use new definition of PublicPseudoIdMask.
* Source/WebCore/rendering/style/RenderStyleSetters.h:
(WebCore::RenderStyle::NonInheritedFlags::setHasPseudoStyles):
- Switch to use new definition of PublicPseudoIdMask.

Canonical link: <a href="https://commits.webkit.org/298737@main">https://commits.webkit.org/298737@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7e9051bf9511c6f2c72302019fab5b73c9cf0e18

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/116499 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/36160 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/26712 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/122556 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/67062 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/118388 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/36854 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/44748 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/88467 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/67062 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/119448 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/29389 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/104505 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/68911 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/28457 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/22612 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/66237 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/98764 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/22772 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/125705 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/43393 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/32582 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/97171 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/43758 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/100714 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/96966 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/42264 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/20173 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/39347 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18602 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/43281 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/48872 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/42747 "Built successfully") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/46087 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/44452 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham reviewed (approved):
> Will be nice to enable this warning everywhere.

This warning will be on by default, but it is only available in newer versions of clang.

Commit:
Fix -Wpreferred-type-bitfield-enum-conversion warnings in WebCore
<https://bugs.webkit.org/show_bug.cgi?id=297439>
<rdar://158245532>

Reviewed by Brent Fulgham.

Warnings found by -Wpreferred-type-bitfield-enum-conversion indicate
that a bitfield used to store an enum type are too small to hold one or
more values defined in the enum.

* Source/WebCore/css/SelectorChecker.cpp:
(WebCore::SelectorChecker::match const):
- Switch to use new definition of PublicPseudoIdMask.
* Source/WebCore/rendering/style/FillLayer.h:
(WebCore::FillLayer::FillBoxBitWidth): Add.
(WebCore::FillLayer::m_clip):
(WebCore::FillLayer::m_origin):
(WebCore::FillLayer::m_clipMax):
- Define FillBoxBitWidth constant for bitwidth of FillBox enum, then
  use it when definining bitfileds of type FillBox.  This fixes the
  warning and a real bug where m_origin and m_clipMax could not hold all
  values of FillBox.
* Source/WebCore/rendering/style/RenderStyleConstants.h:
(WebCore::PseudoId::PublicPseudoIdMask): Remove.
(WebCore::PublicPseudoIdMask): Add.
- Define PublicPseudoIdMask outside enum class PseudoId since it's too
  large to be held by the RenderStyle::pseudoElementType bitfield.  This
  was not a bug in practice, but fixes the warning by moving the mask
  value outside of the enum itself.
* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::RenderStyle::NonInheritedFlags::hasAnyPublicPseudoStyles const):
- Switch to use new definition of PublicPseudoIdMask.
* Source/WebCore/rendering/style/RenderStyleSetters.h:
(WebCore::RenderStyle::NonInheritedFlags::setHasPseudoStyles):
- Switch to use new definition of PublicPseudoIdMask.

Canonical link: https://commits.webkit.org/298737@main
